src/server/api/lib/campaign.ts

```diff
@@ -707,3 +707,36 @@ export const editCampaign = async (
   cacheableData.campaign.reload(id);
   return newCampaign || loaders.campaign.load(id);
 };
+
+export const invalidScriptFields = async (campaignId: string) => {
+  const { rows: variables } = await r.knex.raw(
+    // eslint-disable-next-line no-useless-escape
+    `SELECT regexp_matches(array_to_string(script_options, ','), '\{([^\{]*)\}', 'g') as variable
+FROM interaction_step
+WHERE campaign_id = ?`,
+    [campaignId]
+  );
+
+  const parsedVariables = variables
+    .map((v: { variable: string }) => v.variable)
+    .flat();
+
+  const campaignContact = await r
+    .knex("campaign_contact")
+    .where({ message_status: "needsMessage", campaign_id: campaignId })
+    .orderBy("updated_at")
+    .first();
+  const customFields = Object.keys(JSON.parse(campaignContact.custom_fields));
```

Comment:
Required: Seems like this will error if the campaign has no contacts loaded (seen when trying to create a new campaign).